### PR TITLE
Tidy up repo: ESLint + Prettier, better tooling

### PR DESCRIPTION
**Pull request by tinovyatkin**
This PR is to tidy up 🧹 this repo and make it more inviting for external collaborators. TS/JS has a great tooling and costs nothing to have consistent code style and quality.

There are several things here:

1. Enforcement of your TAB and CRLF policy: it seems from your `.editorconfig` that you want (for some reason) to use TABs and CRLF on your code base. However, it’s not well enforced, so, half of the codebase is violating this requirement (uncomment `linebreak-style` line in new `.eslintconfig.json` and then run `npm run lint` to see). It's hard to keep, particularly for non-Windows contributors (do you really want it this way?).  So, I’ve added more bulletproof settings:

   -   `.gitattributes`
   -   `settings.json` for VSCode
   -   `extensions.json` for VSCode that recommends to install EditorConfig extension
   -   Also, **Prettier** and **ESLint** settings for it (see below)

2. Migrate from deprecated **TSLint** to **ESLint + @typescript-eslint + Prettier**. I made ESLint setting to as closely reflect existing codebase as possible, but once you agreed with settings, we should make a whole run on existing codebase to fix inconsistencies, I assume it better to be done in a separated PR.
Also added ESLint run with JUnit output on Circle CI

3. Upgraded **Typescript** and enabled incremental builds

4. Renamed config files for **Jest** and **TypeDoc** to their _default names_ and drop CLI parameters. Default names for config files simplifies their recognition for a new collaborator and also improves tooling support (such as VSCode JSON scheme support for TypeDoc config and icon for Jest config, etc). I see no reason why do you use a custom name, but may be I’m missing something?

5. Removed unused files from NPM publishing

You have several dependencies in `package.json` that seem to have no use in codebase, except, may be, in mysterious `scratch.ts`. Do you really need `nodemon`, `prompt`, `opn` or `ts-node` for some reason or they just orphaned from previous version?

### CODE REVIEW

**Comment by tinovyatkin** [comment]
Integration tests seems to be broken for quite a while. Is this repo still maintained?

**Comment by iamam34** [comment]
Hi @tinovyatkin 
Firstly, thank you so much - this is beautiful! We agree that this repository should use standard conventions, for many good reasons.
1. We have no strong preference for TAB and CRLF, that's just what we're accustomed to. Happy to accept your recommendation. We want the repository to be accessible for Windows, Mac and Linux users. As a side note, I hadn't heard about VSCode's `extensions.json` until you mentioned it - cool!
2. Upgrading from TSLint to ESLint is a good move. Agree that we should update all line endings in a separate PR.
3. Great!
4. Similar to point 1, there's no real reason that use custom names for these files, it's just bad habit. Thanks for holding us to high standards.
5. Great!

During initial development we used a file called `scratch.ts` to verify contracts against the API. We can remove it from `.gitignore` and `package.json scripts` since it is no longer relevant. Good catch.

Regarding dependencies, it looks like `nodemon` and `opn` are unused. The `prompt` module is referenced by `integration.helpers.ts:readLine()` which is an unused function, so both the dependency and the function can be deleted. The `ts-node` module was referenced in `package.json scripts` to run the scratch file, now obsolete, so that can also be deleted.

We acknowledge the repository hasn't had much attention for a while. We've got some internal discussion around SDK support coming up soon, because we know SDKs are a very important part of the Xero ecosystem and we need to do better.

If you make the suggested changes (standardise line endings, remove scratch.ts) in your fork, I'll review and merge it in next time I'm free. Thanks once again for your PR, much appreciated.

File: src/__integration_tests__/response-body/invoice.response.examples.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export const multipleInvoices: InvoicesResponse = {
 	Id: '431b0b6a-81bf-4286-82f3-73e6f8c78fd3',
 	Status: 'OK',
 	ProviderName: 'Xero API Previewer',
-	DateTimeUTC: '\/Date(1518298927172)\/',
+	DateTimeUTC: '/Date(1518298927172)/',
```
Comment (iamam34) [inline src/__integration_tests__/response-body/invoice.response.examples.ts]: Are you sure about this? The API really does return a string with an escaped backslash. as described in https://developer.xero.com/documentation/api/requests-and-responses.

Comment (tinovyatkin) [reply src/__integration_tests__/response-body/invoice.response.examples.ts]: That's actually are automatic fix by ESLint, but yes, I'm pretty sure backslash doesn't need to be escaped here to be threatened as backslash.

**Comment by SerKnight** [comment]
Maintenance of this repo will be picked back up soon - will refactor these changes into OAuth2  work soon as I can!

**Pull request merged**